ribs/archives/_archive_base.py

```diff
@@ -435,21 +513,23 @@ def add(self,
 
         # Copy old objectives since we will be modifying the objectives storage.
         old_objective_batch = np.copy(self._objective_arr[index_batch])
+        old_threshold_batch = np.copy(self._threshold_arr[index_batch])
 
         # Compute the statuses -- these are all boolean arrays of length
         # batch_size.
         already_occupied = self._occupied_arr[index_batch]
         is_new = ~already_occupied
         improve_existing = (objective_batch >
-                            old_objective_batch) & already_occupied
+                            old_threshold_batch) & already_occupied
         status_batch = np.zeros(batch_size, dtype=np.int32)
         status_batch[is_new] = 2
         status_batch[improve_existing] = 1
 
-        # Since we set the new solutions in old_objective_batch to have
+        # Since we set the new solutions in old_threshold_batch to have
         # value 0.0, the values for new solutions are correct here.
         old_objective_batch[is_new] = 0.0
-        value_batch = objective_batch - old_objective_batch
+        old_threshold_batch[is_new] = 0.0
```

Comment:
As noted for `add_single`, I think we should only do this if the threshold_min if -np.inf. Otherwise, we should compute with respect to the threshold_min.